api/include/opentelemetry/context/threadlocal_context.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include "opentelemetry/context/context.h"
+#include "opentelemetry/context/runtime_context.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+// The ThreadLocalContext class is a derived class from RuntimeContext and
+// provides a wrapper for propogating context through cpp thread locally.
+// This file must be included to use the RuntimeContext class if another
+// implementation has not been registered.
+class ThreadLocalContext : public RuntimeContext
+{
+public:
+  ThreadLocalContext() = default;
+
+  // Return the current context.
+  Context InternalGetCurrent() override { return stack_.Top(); }
+
+  // Resets the context to a previous value stored in the
+  // passed in token. Returns true if successful, false otherwise
+  bool InternalDetach(Token &token) override
+  {
+    if (!(token == stack_.Top()))
+    {
+      return false;
+    }
+    stack_.Pop();
+    return true;
+  }
+
+  // Sets the current 'Context' object. Returns a token
+  // that can be used to reset to the previous Context.
+  Token InternalAttach(Context context) override
+  {
+    stack_.Push(context);
+    Token old_context = CreateToken(context);
+    return old_context;
+  }
+
+private:
+  // A nested class to store the attached contexts in a stack.
+  class Stack
```

Comment:
Let's mark all `Stack` methods as `noexcept`. `Top` can be `const` too.

Reply:
Marked as noexcept and const.